### PR TITLE
Add recording feature to websockify.js.

### PR DESCRIPTION
**Pull request by darkpixel**
See PR #284

### CODE REVIEW

File: other/js/websockify.js [hunk 1]
```diff
@@ -26,20 +26,36 @@ var argv = require('optimist').argv,
 
 
 // Handle new WebSocket client
-new_client = function(client) {
+new_client = function(client, req) {
```
Comment (CendioOssman) [inline other/js/websockify.js]: What happened here? This is already on `master`?

File: other/js/websockify.js [hunk 2]
```diff
@@ -26,20 +26,36 @@ var argv = require('optimist').argv,
...
+    var start_time = new Date().getTime();
+
+    console.log(req ? req.url : client.upgradeReq.url);
     log = function (msg) {
         console.log(' ' + clientAddr + ': '+ msg);
     };
     log('WebSocket connection');
     log('Version ' + client.protocolVersion + ', subprotocol: ' + client.protocol);
 
+    if (argv.dir) {
```
Comment (CendioOssman) [inline other/js/websockify.js]: Is this really correct? The argument is `--record-dir`.

**Comment by CendioOssman** [comment]
PS. You don't have to open a new PR when you want to change things. Just push to the same branch and everything will be updated.

**Comment by darkpixel** [comment]
Yeah--I apparently hosed the PR up.
I'm not sure how to take a branch that already has multiple commits and squash it unless I'm merging into another branch (i.e. the --squash flag or the option in GitHub to merge and squash).   I'll clean get it cleaned up.

**Comment by darkpixel** [comment]
Weird.  I've been using git for about 9 years now, didn't realize that was part of rebase. ;)

**Comment by CendioOssman** [comment]
Are you able to sort things out so we can merge this? :)

**Comment by darkpixel** [comment]
Sorry--it fell off my radar (TabWrangler for Chrome is both wonderful and terrible).

I'll jump back in after work. :)

**Comment by darkpixel** [comment]
I've never used rebase to intentionally lose commit history, but I think I've got it now. ;)

**Pull request merged**